### DISCUSSION TITLE
Initialize and increment at the same time

### DISCUSSION THREAD

**Discussion by gustavopch** [DanielXMoore/Civet#315] [Ideas]
The idea is to have a shorthand to initialize a property and increment/decrement it at the same time. So basically a shorthand for this:
```ts
obj[key] ??= 0
obj[key] += 1
```

## Syntax 1 (can't specify the initial value unfortunately)
```ts
obj[key] ?+= 1
```

## Syntax 2 (uses a pipeline)
```ts
obj[key] ?= 0 |>+= 1
```

## Syntax 3 (combines two operators in the same line)
```ts
obj[key] ?= 0 += 1
// Or, alternatively:
obj[key] ?? 0 += 1
```

Might be relevant to consider providing a shorthand for other similar cases like `(array ?= []).push(item)`.

**Comment by edemaine** [comment]
Syntax 3 is problematic because it already has a meaning:

```js
obj[key] ?= x += 1    ====    obj[key] ?= (x += 1)
obj[key] ?? x += 1    ====    obj[key] ?? (x += 1)
```

But we could make the following work:

```js
(obj[key] ?= x) += 1    ====    obj[key] ?= x, obj[key] += 1
```

**Reply by edemaine** [reply]
Update: `(obj[key] ?= x) += 1` [now works](https://civet.dev/playground?code=KG9ialtrZXldID89IHgpICs9IDE%3D), and once #353 gets merged, so will `(obj[key] ?= x)++`.  So Syntax 3 is basically finished.

We could still pursue the other syntaxes.  In particular, I think Syntax 2 is a nice extension to pipe assignment operators:
* `|>+=` and other update operators, beyond the existing `|>=`
* Support for assignment and update operations in LHS of pipe assignment (similar to #348 and #353 but for pipes)
